### ISSUE TITLE
Unable to add books to shelf

### ISSUE THREAD

**Issue by jhofker** [janeczku/calibre-web#3210]
**Describe the bug/problem** 

I have to imagine this is somehow my fault since I see no other reports of this behavior, but I simply cannot add books to shelves, nor can I save edits to users.

When I click the buttons to perform these actions, nothing happens. No network request is made (and thus no network requests fail).

**To Reproduce**

1. Create a shelf
2. Go to books
3. Click any book
4. Scroll down and attempt to add the book to a shelf

Or:
1. Click "Admin"
2. Click on a user in the list
3. Change the user's email address
4. Scroll down and click save

**Logfile**

Nothing relevant in it.

**Expected behavior**

A list of shelves appears so that the book can be added

The user is edited

**Screenshots**

**Environment (please complete the following information):**

 - OS: [e.g. Windows 10/Raspberry Pi OS]
 - Calibre-Web version: 0.6.23
 - Docker container: LinuxServer
 - Browser: Tried in Firefox and Chrome

**Additional context**
Accessing via a reverse proxy

**Comment by OzzieIsaacs** [comment]
I suspect some issues with the reverse proxy configuration.

Could you please try to access calibre-web directly and check if it's working in this case?

Did you check the developer console (Firefox hotkey: ctrl+Shift+k, chrome hotkey: ctrl+shiift+i) for requests? 
Are any javascript errors showing up?

Are you using the standard them, or the caliblur! Theme (try to use the standard theme)?

**Comment by jhofker** [comment]
Hey Ozzie, thanks for the quick reply!

> **tl;dr: Remove traefik compression middleware for calibre-web.**

- No errors in the console
- No non-200 responses to requests
- Default theme

I share your suspicion. I was experimenting with it last night to see what the differences are and here's the trail I went down.

Comparing reverse-proxied (https://calibre.local.domain.tld/admin/usertable) to direct side by side yields three additional network requests on the direct:

<img width="591" alt="image" src="https://github.com/user-attachments/assets/c1c214ac-2a09-4ba8-85af-4cac031e8bcb">

Okay but why can't jquery make those requests? Well on the reverse-proxied side, all js resources that it tries to load come in as 0 bytes:
<img width="751" alt="image" src="https://github.com/user-attachments/assets/3870f09f-fc00-4b31-b8f9-553da5d29914">

That seems weird, why would the reverse proxy do that? I don't have a good answer. Let's update the reverse-proxy (traefik) in case there's a bug. Loading on 3.2.0 now, we see slightly different results:
<img width="708" alt="image" src="https://github.com/user-attachments/assets/ab794e7d-2cac-480b-94ad-47325601def8">

Okay, better errors are good I suppose, but that still doesn't explain things. Maybe it's just a Firefox thing? No, here's Chrome:
<img width="687" alt="image" src="https://github.com/user-attachments/assets/04da69e4-679d-4f17-99d7-ade03ef734a8">

Is it an HTTP2 thing? Let's try requesting a file with curl (right-click on the request, copy as curl) but pass the http2 flag to it:
```
> curl --http2 'https://calibre.local.domain.tld/static/js/libs/jquery.min.js' \
  -H 'accept: */*' \
  -H 'accept-language: en-US,en;q=0.9' \
  -H 'cookie: remember_token=... \
  -H 'priority: u=2' \
  -H 'referer: https://calibre.local.domain.tld/' \
  -H 'sec-ch-ua: "Chromium";v="130", "Google Chrome";v="130", "Not?A_Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: script' \
  -H 'sec-fetch-mode: no-cors' \
  -H 'sec-fetch-site: same-origin' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36'

/*! jQuery v3.6.3 | (c) OpenJS Foundation and other contributors | jquery.org/license */
!function(e,t){"use strict";"object"==typeof module&&"object"==typeof module.exports?module.exports=e.document?t(e,!0):function(e){if(!e.document)throw new Error("jQuery requires a window with a document");return t(e)}:t(e)}("unde....
```

Nope that's all there.

Why can the browser request assets like css but not js? What's different there? Only difference in the requests that I can see are the `Accept` header. CSS and fonts have a specified MIME type and js files will take anything, but that should work just fine and the direct access method shows the correct response headers.

Maybe there's a reverse proxy setting I need to change? Looking at [reverseproxy.py](https://github.com/janeczku/calibre-web/blob/master/cps/reverseproxy.py), it doesn't seem like it. And that doesn't make too much sense anyway -- curl works against the url but the browser doesn't. This usually means something fun about CORS...

So what's the CORS implementation? Hmm, security headers [get added](https://github.com/janeczku/calibre-web/blob/bf9313ddeb0e2a821186ffe4ff44c6da717e34a9/cps/web.py#L90), but `'self'` is set and that's all we're doing here. I'll add my reverse-proxy url anyway on the `/admin/config` page. After doing that, I see the url coming back when .js files are requested via direct acces, but the same errors persist when proxied.

In the docs for reverse proxies, I see [an example for kobo sync](https://github.com/janeczku/calibre-web/wiki/Setup-Reverse-Proxy#traefik-260-example-kobo-sync-fix-enabled) that sets the `X-Scheme` header on requests. No change after doing this.

While I was in the settings page, I enabled the debug and access logs to see if they would tell us something, but debug logs show nothing relevant and the access logs show successful requests for the .js assets:
```
10.0.10.58 - - [2024-10-30 10:45:07] "GET /static/js/libs/intention.min.js HTTP/1.1" 200 6449 0.000825
10.0.10.58 - - [2024-10-30 10:45:07] "GET /static/js/libs/context.min.js HTTP/1.1" 200 2116 0.000788
10.0.10.58 - - [2024-10-30 10:45:07] "GET /static/js/libs/plugins.js HTTP/1.1" 200 16964 0.000830
10.0.10.58 - - [2024-10-30 10:45:07] "GET /static/js/libs/jquery.form.min.js HTTP/1.1" 200 16971 0.000793
10.0.10.58 - - [2024-10-30 10:45:07] "GET /static/js/uploadprogress.js HTTP/1.1" 200 7986 0.000697
10.0.10.58 - - [2024-10-30 10:45:07] "GET /static/js/main.js HTTP/1.1" 200 16960 0.000809
```

Okay...so we're back to it being a browser + traefik type of issue. Searching around, I see [this issue](https://github.com/janeczku/calibre-web/issues/3045) and that's backed up by [this thread](https://community.traefik.io/t/calibre-web-behind-traefik-working-but-ui-and-too-many-requests-issue/22928/22). I changed my traefik config to remove compression, restarted things and...

**Now it works.**
